Fix #399 (fussy popover target)

SabreCat opened the pull request:
Some shuffling of task popover functionality to address the difficulty of targeting the "notes" speech bubble icon.
- The icon itself no longer has any popover. It simply cues the user that more detail is available.
- Popovers now appear when you mouse over the task text. For equipment Rewards, mousing over the shop sprite also works.
- Implemented and tested for tasks, static rewards, spells, and Winter special spells.
- Tested with drag-drop sort operations. Looks a little funky, but works fine.

SyntaxPickax committed:
Fixed challenge prize logic to avoid double paying

Commit:
Fixes #1959: petCount was only being updated...

in groupsCtrl.js. This made profile.petCount blank unless the current
user viewed himself in chat. I put the code to set profile.petCount into
userCtrl.js.

Note: petCount (not profile.petCount) is still always equal to the
current user's pet count. There is probably a better solution than this,
I just don't know enough about AngularJS to make it.

SabreCat committed:
Fix #399 (fussy popover target)

SabreCat committed:
Fix #399 (fussy popover target)

lefnire commented:
**This** is what I meant to reply to (sorry)

> I'm not so sold in this. I know it's the lesser of two evils, but let me sit on it for a while - there might be a better way. 'Course, if I'm sitting too long let's just merge, and we can find the better way later

SabreCat commented:
Yeah, if there's a better solution I'm all for it. Somebody (once again, I've lost track of who) suggested having the popover on the task title instead of the icon, so I decided to take it for a spin. Take or leave!

lefnire committed:
quest: when new user joins party, invite to pending-invite quests. When
user in quest leaves party, knock user from group.quest.members & quest
from user.party.quest

lefnire committed:
fix #2155 . Since quest.leader is new, doesn't apply to quests started
before today. Show abort/begin for all quest members if quest.leader
doesn't exist, otherwise only show those options to the leader

lefnire committed:
Merge pull request #1984 from SyntaxPickax/patch-2

Fixed challenge prize logic to avoid double paying

lefnire committed:
Merge pull request #2153 from HabitRPG/sabe/tweak3

Don't need angle brackets anymore for party cast notifications

lefnire committed:
Merge pull request #1986 from Magnilucent/develop

Fixes #1959: petCount was only being updated...

lefnire commented:
Ok, I dig it. I'm wondering, looks like your'e putting the popover on the text as well as shop icons. Fair bit of duplication in multiple places, did it not work well with the popover on the full task element, top-level? (that make sense?)

lefnire committed:
#1499 ng-if to hide/show task editing (fewer watchers)

lefnire committed:
#1499 add list filter on tasks to not display completed todos at all
 until "completed" tab clicked

lefnire committed:
#1499 check that task[0] exists first

lefnire committed:
add markdown support for task.text so we can use icons in task
titles. @wc8 does this satisfy
https://trello.com/c/FCVdjdUd/102-task-reward-icons sufficiently? @lemoness

lefnire committed:
display profile.stats.class in modal

lefnire committed:
task-icons: larger emoji in task text than chat

SabreCat commented:
@lefnire I actually hadn't thought of that, silly me. Will it be awkward to have the popover show up simultaneously with other tooltips, like Challenge/Edit/Progress icons? Worth a try in any case, I'll check it out and send a new commit in if it looks OK.

SabreCat committed:
Put popover text at <li> level instead of <span>

SabreCat committed:
Fix incorrect placement of popover in Spells

SabreCat committed:
Merge branch 'sabe/tweak' of https://github.com/HabitRPG/habitrpg into sabe/tweak

SabreCat commented:
@lefnire 

Oh lordy. :scream_cat: When trying to test this, I couldn't check off tasks anymore given my habitrpg-shared branch having gotten ahead of my habitrpg. So I attempted a rebase of my changes onto develop, a harrowing experience. I think I figured out how to deal with the merge conflicts, and things seem OK locally, but all those extra commits scare me.

So, notables:
- Commits 7d5b639 and 81ab6cd are the ones implementing [your suggestion](https://github.com/HabitRPG/habitrpg/pull/2083#issuecomment-31311722).
- As for the testing, everything looks pretty good with the popovers being on the `<li>` elements instead of the task-text `<span>`. One exception: the Rewards column stretches by a few pixels when you get the popover for the bottommost element of a section (e.g. Health Potion); no idea how to fix that.
- You'd better give this a very close look, since it's the first time I've done conflict resolution.
- Advice for next time? Should I have done a straight merge instead of a rebase, maybe? Reset my habitrpg-shared branch to the same point in time as my habitrpg branch's changes? (The culprit appeared to be the "progress" bit used by Bosses, which I didn't have.) Or was this the right way to go, however cluttered?

SabreCat commented on views/shared/tasks/lists.jade:
```diff
@@ -58,13 +58,13 @@ script(id='templates/habitrpg-tasks.html', type="text/ng-template")
                 span.shop_gold
             // main content
             span(bo-class='{"shop_{{item.key}} shop-sprite item-img": true}')
-            p.task-text {{item.text}}
```
Aw shoot, I thought I'd resolved these conflicts in favor of `p` instead of `span`. Whichever it's supposed to be, keep that.

SabreCat commented:
@lefnire, if you just want me to do this over, I can. It's a pretty minor change to come with such a pile of merge commits. (But if you do have tips for future reference per [above](https://github.com/HabitRPG/habitrpg/pull/2083#issuecomment-31318193), I'd still appreciate 'em!)

lefnire commented:
@SabreCat looks like this branch got a bit borked (rebases can get crazy). How about a fresh start with this commit: https://github.com/HabitRPG/habitrpg/commit/50d241e2c801425ba7e4c6f124b0ac1b156697d0 ? This look like I covered everything?

SabreCat commented:
Yeah, that looks right! Thanks for cleaning up my mess :cat::sweat_drops:

SabreCat commented:
Looks like you've got your own branch ready to roll for this one, so I'll close this PR.
